include/tc/core/cpu/cpu_tc_executor.h

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <dlpack/dlpack.h>
+
+#include "tc/core/halide_utils.h"
+#include "tc/core/mapping_options.h"
+#include "tc/core/polyhedral/scop.h"
+#include "tc/core/tc_executor.h"
+#include "tc/core/utils/dlpack.h"
+#include "tc/lang/parser.h"
+
+namespace tc {
+
+struct CpuRTCFunction {
+  void clear() {}
+};
+
+class CpuTcExecutor : public ::tc::TcExecutor {
+ public:
+  CpuTcExecutor(
+      std::string id,
+      const std::vector<const DLTensor*>& inputsInfo,
+      const std::string& options,
+      lang::TreeRef tcDefinition)
+      : TcExecutor(id, inputsInfo, options, tcDefinition) {}
+
+  ~CpuTcExecutor() {}
+
+  CpuTcExecutor(CpuTcExecutor&&) = delete;
+  CpuTcExecutor& operator=(CpuTcExecutor&&) = delete;
+  CpuTcExecutor(const CpuTcExecutor&) = delete;
+  CpuTcExecutor& operator=(const CpuTcExecutor&) = delete;
+
+  // Can only be called once with specific kernel options.  Input sizes are
+  // set up as constructor argument and output sizes are inferred.
+  //
+  // If you need another kernel for another Tc or another inputs, outputs,
+  // options then just instantiate another CpuTcExecutor.
+  // This is because for the time being we fully specialize all the sizes and
+  // strides at runtime.
+  // @{
+  void compile(const std::string& options) override {
+    compile(MappingOptions(options));
+  }
+  void compile(const tc::MappingOptions& options);
```

Comment:
I wouldn't reuse `MappingOptions` here, most of them are gpu-specifc, except perhaps tile sizes.

Reply:
Right, so should we start redesigning the mapping options to have the core and the target specific ones clearly separated ? Same thing, depending on urgency I can do here or in a follow up PR.
@salexspb please lmk if you can wait a bit more of if you need this merged. I can probably address @ftynse deeper comments by tomorrow night PST.